earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -120,22 +120,15 @@ void PacketGenAssetVersionImplD::UpdateChildren(
     prevTime = curTime;
 
     // Determine which insets intersect us at this level
-    const khLevelCoverage levCov(config.coverage.levelCoverage(level));
+    khLevelCoverage levCov(config.coverage.levelCoverage(level));
     std::vector<uint> neededIndexes;
+
     neededIndexes.reserve(numInsets);
-    if (type == AssetDefs::Imagery) {
-      FindNeededImageryInsets(khInsetCoverage(levCov),
-                              extra.insetInfo, numInsets,
-                              neededIndexes,
-                              config.beginMinifyLevel,
-                              config.endMinifyLevel);
-    } else {
-      FindNeededTerrainInsets(khInsetCoverage(levCov),
-                              extra.insetInfo, numInsets,
-                              neededIndexes,
-                              config.beginMinifyLevel,
-                              config.endMinifyLevel);
-    }
+    auto gencov = khInsetCoverage(levCov);
+
+    extra.calc->setBeginMinifyLevel(config.beginMinifyLevel);
+    extra.calc->setEndMinifyLevel(config.endMinifyLevel);
+    neededIndexes = std::move(extra.calc->GetOverlapForLevel(gencov));
```

Comment:
I just realized that the `extra.insetInfo` variable is different here than it is in BuildIndex. I think this will work but it may take longer. I think the most efficient way to do this is to pass in the entire envelope each time to make sure that we get the right inputs.